src/main/java/com/amazon/opendistroforelasticsearch/sql/plugin/RestSqlAction.java

```diff
@@ -91,6 +95,11 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
                 return channel -> restExecutor.execute(client, additionalParams, queryAction, channel);
             }
         } catch (Exception e) {
```

Comment:
ideally this should be
```java
catch (final NullPointerException | SqlParseException | ParserException | SQLFeatureNotSupportedException | IllegalArgumentException | IndexNotFoundException e) {
  // do whatever
} catch (final Exception e) {
  // handle generic case
}
```

but even if you want to keep it this way please update to this:
```java
Metrics.getInstance().getNumericMetric(isClientError(e) ? MetricType.FAILED_REQ_COUNT_CUS : MetricType.FAILED_REQ_COUNT_SYS).increment();
```

Reply:
Also, I know it's not your code, but can you fix `isClientError` method to use `||` instead of `|` .

Reply:
> Also, I know it's not your code, but can you fix `isClientError` method to use `||` instead of `|` .

Good catch. That was my mistake. :)

Reply:
Good catch! Thanks.